src/ui/CodingContractPopup.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react";
+import { KEY } from "../../utils/helpers/keyCodes";
+
+import { CodingContract, CodingContractType, CodingContractTypes } from "../CodingContracts";
+import { ClickableTag, CopyableText } from "../ui/React/CopyableText";
+import { PopupCloseButton } from "../ui/React/PopupCloseButton";
+
+type IProps = {
+    c: CodingContract;
+    onCloseClick: () => void;
```

Comment:
To match naming convention these should be named `onClick` and `onSolve`.

also it's an `attempt` not a `solve`